operator/podinfo/podinfo_controller_suite_test.go

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package podinfo
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var (
+	cfg     *rest.Config
+	ctx     context.Context
+	cancel  context.CancelFunc
+	testEnv *envtest.Environment
+)
+
+type ControllerTestSuite struct {
+	suite.Suite
+	k8sClient  client.Client
+	Reconciler *Reconciler
+}
+
+// TODO: Remove fmt.Println statements with the logging.
+// TODO: Enhance the Makefile to delete the test-podinfo cluster afterwards.    - Bootstrap test environment
```

Comment:
please address these TODOs 🙏

Reply:
yes, I am almost done with the logging, just haven't pushed it because it's not clean enough yet. I should do it first.

Reply:
done